templates/CRM/common/paymentBlock.tpl

```diff
@@ -85,9 +85,11 @@
       var $form = $('#billing-payment-block').closest('form');
       {/literal}
       {if $contributionPageID}
-        {capture assign='contributionPageID'}id={$contributionPageID}&{/capture}
+        {capture assign='pageID'}id={$contributionPageID}&module=CiviContribute&{/capture}
```

Comment:
Hmm - I think we should either pass entity+entityID or the specific field required. 

I tend to prefer the latter just because in CiviCRM people tend to make use of any parameter they get in wierd unpredictable ways and I like to limit people's options when it comes to 'tacking on functionality' - but I could probably go either way on it.